src/include/ipc/topology.h

```diff
@@ -83,11 +83,20 @@ struct sof_ipc_comp {
 #define SOF_MEM_CAPS_CACHE			(1 << 6) /**< cacheable */
 #define SOF_MEM_CAPS_EXEC			(1 << 7) /**< executable */
 
+#define SOF_BUF_FLAG_OVERRUN_NO_XRUN	(1 << 0) /**< overrun will cause ring
+						  * buffer overwrite, instead
+						  * of XRUN */
+#define SOF_BUF_FLAG_UNDERRUN_NO_XRUN	(1 << 1) /**< underrun will cause
+						  * readback of 0s, instead of
+						  * XRUN */
+
 /* create new component buffer - SOF_IPC_TPLG_BUFFER_NEW */
 struct sof_ipc_buffer {
 	struct sof_ipc_comp comp;
 	uint32_t size;		/**< buffer size in bytes */
 	uint32_t caps;		/**< SOF_MEM_CAPS_ */
+	uint32_t flags;		/**< SOF_BUF_FLAG_ */
+	uint32_t reserved;	/**< reserved for future use */
```

Comment:
Just one thought here, is this structure contained by another ? Or is it sent stand alone ?

Reply:
AFAIK, it's standalone. At least that what I've found when digging through the interface on FW side, but I would really appreciate if some1 could doublecheck that.

Reply:
@akloniex CI is telling us it is good. I would expect this to go all red if we were contained.